persistence/persistence-api/src/main/java/org/cbioportal/persistence/CacheEnabledConfig.java

```diff
@@ -1,13 +1,30 @@
 package org.cbioportal.persistence;
 
 import org.springframework.beans.factory.annotation.Value;
+import java.util.ArrayList;
+import java.util.Arrays;
 
 public class CacheEnabledConfig {
 
     private boolean enabled;
+    
+    public static final String DISK = "disk";
+    public static final String HEAP = "heap";
+    public static final String HYBRID = "hybrid";
 
-    public CacheEnabledConfig(String cacheEnabled) {
-        this.enabled = Boolean.parseBoolean(cacheEnabled);
+    public static ArrayList<String> validCacheTypes = new ArrayList<String>(Arrays.asList(DISK, HEAP, HYBRID));
+
+    public CacheEnabledConfig(String cacheType) {
+        this.enabled = enableCache(cacheType);
+    }
+
+    public static boolean enableCache(String cacheType) {
+        for (String validCacheType : validCacheTypes) {
```

Comment:
I think we could do this with `validCacheTypes.contains(cacheType.toLowerCase())` ... searching an array of length 3 is fine with a linear scan (rather than hashing in a set)

Reply:
`validCacheTypes.contains(cacheType.toLowerCase())` would throw a null pointer exception because cacheType can be null.  That is what was breaking the Travis tests.  We could of course get around that... but we don't really need to.

Reply:
👍